import_export/results.py

```diff
@@ -106,6 +106,7 @@ def append_base_error(self, error):
         self.base_errors.append(error)
 
     def add_dataset_headers(self, headers):
+        headers = list() if not headers else headers
```

Comment:
I had a look at where this function is used and it's used exactly one place; maybe it would be better for the consumer of the function to control what it puts in? It feels to me like it's fairly pythonic to have a function blow up if it's given bad data (trying to set the headers to None). 

```
        if collect_failed_rows:
            result.add_dataset_headers(dataset.headers)
```
Maybe you could say 'if collect_failed_rows and dataset.headers' ?

Reply:
My thinking is that none of the subsequent code for trying to parse stuff will run if there is an empty dataset, but could be underthinking that one. 

Maybe there's a clean way to report that the dataset is empty.